dali/operators/generic/transpose/transpose.h

```diff
@@ -16,37 +16,82 @@
 #define DALI_OPERATORS_GENERIC_TRANSPOSE_TRANSPOSE_H_
 
 #include <algorithm>
+#include <functional>
 #include <memory>
 #include <vector>
 
-#include "dali/pipeline/operator/operator.h"
+#include "dali/kernels/common/utils.h"
 #include "dali/operators/generic/transpose/cutt/cutt.h"
+#include "dali/pipeline/operator/operator.h"
 
 namespace dali {
 
 namespace detail {
 
-template <typename T>
-T Permute(const T& in, const std::vector<int>& permutation) {
-  T out = in;
-  for (size_t i = 0; i < permutation.size(); i++) {
-    auto idx = permutation[i];
-    out[i] = in[idx];
+template <typename ShapeT>
+inline void RowToColumnMajor(ShapeT *dims, int *perm, size_t len) {
```

Comment:
```suggestion
inline void RowToColumnMajor(ShapeT *dims, span<int> perm) {
```

Reply:
That would indicate that the spans can have different length IMO.

Reply:
Using two spans would make it possible to pass different lengths to the function.